src/components/settings/SettingsSSOTab/index.jsx

```diff
@@ -115,41 +117,44 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
           </div>
         </div>
         {(!isLoading || !pdIsLoading) && (
-          <div>
-            {/* providerConfig represents the currently selected config to edit/create, if there are
-            existing configs but no providerConfig then we can safely render the listings page */}
-            {existingConfigs?.length > 0 && (providerConfig === null) && (
-              <NewExistingSSOConfigs
-                providerData={existingProviderData}
-                configs={existingConfigs}
-                refreshBool={refreshBool}
-                setRefreshBool={setRefreshBool}
-              />
+          <>
+            {!error && (
+              <div>
+                {/* providerConfig represents the currently selected config to edit/create, if there are
+                existing configs but no providerConfig then we can safely render the listings page */}
+                {existingConfigs?.length > 0 && (providerConfig === null) && (
+                  <NewExistingSSOConfigs
+                    providerData={existingProviderData}
+                    configs={existingConfigs}
+                    refreshBool={refreshBool}
+                    setRefreshBool={setRefreshBool}
+                    setPollingNetworkError={setPollingNetworkError}
+                  />
+                )}
+                {/* Nothing found so guide user to creation/edit form */}
+                {showNoSSOCard && (
+                  <NoSSOCard setShowNoSSOCard={setShowNoSSOCard} setShowNewSSOForm={setShowNewSSOForm} />
+                )}
+                {/* Since we found a selected providerConfig we know we are in editing mode and can safely
+                render the create/edit form */}
+                {((existingConfigs?.length > 0 && providerConfig !== null) || showNewSSOForm) && (<NewSSOConfigForm />)}
+                {pdError && (
+                <Alert variant="warning" icon={WarningFilled}>
+                  An error occurred loading the SAML data: <p>{pdError?.message}</p>
+                </Alert>
+                )}
+                <Toast
+                  onClose={() => setInfoMessage(null)}
+                  show={infoMessage.length > 0}
```

Comment:
We're setting `infoMessage` to null when closing above.
```suggestion
                  show={infoMessage?.length > 0}
```